lib/mpl_toolkits/axisartist/axislines.py

```diff
@@ -124,9 +124,7 @@ def __init__(self, loc, nth_coord=None):
                 {"bottom": 0, "top": 0, "left": 1, "right": 1}, loc=loc))
         if (nth_coord == 0 and loc not in ["left", "right"]
                 or nth_coord == 1 and loc not in ["bottom", "top"]):
-            _api.warn_deprecated(
-                "3.7", message=f"{loc=!r} is incompatible with "
-                "{nth_coord=}; support is deprecated since %(since)s")
+            raise ValueError("x=None is not supported")
```

Comment:
Where did this error message come from? It seems unrelated to the checks that are guarding it, what is `x`?

I think the message should likely be `f"{loc=!r} is incompatible with {nth_coord=}"`